### ISSUE TITLE
Colors of sender names in group chats differ from member list

### ISSUE THREAD

**Issue by ttlmax** [hoehermann/purple-signald#69]
While the entries in a group chat's member list are shown in different colors, the sender's name in the chat messages are all shown in the same color (some dark yellow for me).

I have no idea why pidgin is probably not able to get the color for the sender of the message from the related user in the members list as it is the case in chats with other protocols. Is there something that is missing in the plugin?

**Comment by hoehermann** [comment]
I confirm that _outgoing_ and synced messaged are displayed in arbitrary colors. 

For _incoming_ messages however, it looks stable to me. I added all the people I chat with to my buddy list. Maybe that is needed for some mapping.

Though my options to test this behaviour are rather limited. I only have one phone and one emulated device. Quite possible that it looks okay if there are just two people in the chat room. :/

**Comment by ttlmax** [comment]
That's interesting. I can see different colors for the buddies (and the message time) in ordinary conversation (incoming name red, outgoing name blue). But in group chats, all names are displayed in some ochre yellow, ignoring the buddy's color in the member list

**Comment by hoehermann** [comment]
![color](https://user-images.githubusercontent.com/838331/163860703-c54b13ea-1f19-41d8-af91-ce668912b99c.png)
"Works for me." 🤷

**Comment by ttlmax** [comment]
Indeed, that looks as expected, thanks for the screenshot. Then I'll have to go through my preferences and check which of them might change the colors as a side effect.

**Comment by ttlmax** [comment]
Found the issue. The color of each user in a group chat turns to ochre yellow when the option _Receive notifications for all group messages_ is activated. This functionality is implemented by faking that the user's nick is mentioned in each incoming message. Obviously this also changes the color of the sender's name.

**Comment by hoehermann** [comment]
Nice catch. This feature was introduced by 296016d. I never used it myself. Apart from this side-effect, does the feature work as intended for you?

**Comment by ttlmax** [comment]
I saw this in another plugin (I think it was matrix). It makes pidgin's tray icons to indicate a new message which is otherwise not the case for chats. But I am really wondering if this feature is sufficiently important compared to loosing user name colors in chat windows.

**Comment by hoehermann** [comment]
By default, the feature is disabled. I want to document the side-effect in the readme. Everything else is an end-user-decision. :)

**Comment by hoehermann** [comment]
This issue has arisen again in the dev-branch. Affects outgoing messages (sent locally or remotely and synced). Incoming messages from others seem to be fine.

**Comment by hoehermann** [comment]
It is now possible to use your UUID as a username in Pidgin. This way, everyone is identified by their UUID, including the local user. Although unwieldy, this seems to resolve the problem.
